File: hw/codena/u1/hw02chickenSimulationInC/Python/view/chicken_simulator.py
from model.chicken import Chicken

chicken = Chicken("Lucy", "Brown and White", 2, False, 0, 0)

option = 1

while(option != 8):

    print("\n---Actions---")
    print("1. Show information")
    print("2. Cluck")
    print("3. Wander")
    print("4. Eat")
    print("5. Drink")
    print("6. Poop")
    print("7. Lay an egg")
    print("8. Exit")

    try:

        option = int(input("Select an option: "))

        match option:

            case 1:
                chicken.show_information()

            case 2:
                chicken.cluck()

            case 3:
                chicken.wander()

            case 4:
                chicken.eat()

            case 5:
                chicken.drink()

            case 6:
                chicken.poop()

            case 7:
                chicken.lay_an_egg()

            case 8: 
                print("Finish")

            case _:
                print("Invalid option")
 
    except ValueError:
        print("Enter a valid number")

File: hw/codena/u1/hw02chickenSimulationInC/JavaScript/view/ChickenSimulator.js
import Chicken from "../model/Chicken.js";
import PromptSync from "prompt-sync";

const prompt = new PromptSync();

let chicken = new Chicken(
    "Lucy",
    "Brown and White",
    2,
    false,
    0,
    0
)

let option = 1;

do {

    console.log("\n---Actions---");
    console.log("1. Show information\n2. Cluck\n3. Wander");
    console.log("4. Eat\n5. Drink\n6. Poop");
    console.log("7. Lay an egg\n8. Exit");

    try{

        option = Number(prompt("Select an option: "));

        switch(option){

            case 1:
                chicken.showInformation();
                break;

            case 2:
                chicken.cluck();
                break;

            case 3:
                chicken.wander();
                break;

            case 4: 
                chicken.eat();
                break;

            case 5:
                chicken.drink();
                break;
            
            case 6:
                chicken.poop();
                break;
            
            case 7:
                chicken.layAnEgg();
                break;
            
            case 8:
                console.log("\nFinish");
                break;

            default:
                console.log("\nIncorrect option.");
        }

    } catch (error){
        console.log("\nEnter a valid number.")
    }
    
    
} while (option != 8);
